tdp/core/collections.py

```diff
@@ -46,6 +47,9 @@ def __iter__(self):
     def __len__(self):
         return self._collections.__len__()
 
+    def items(self) -> Iterator[Tuple[str, Collection]]:
```

Comment:
Why do you implement this method ? It should already be implemented because `Collections` extends `Mapping` no ?

Reply:
To have the correct type inferred.

For example:

```py
for collection_name, collection in collections.items():
```

By default, `collection_name` and `collection` are both of type `Any`. Overriding the `Mapping.items` methods allows to specifies the returned types.

Reply:
Just for type, seriously ...

Reply:
I haven't find a better way to do it 🤷🏻‍♂️

Reply:
And you will do the same for all mixing method of `Mapping` like `keys`, `values` and `get` ? We lose the value of `Mapping` "abc" to only implement required method.

Reply:
Well, this wouldn't be a problem with Python 3.9: https://docs.python.org/3.9/library/typing.html#generics